internal/util/tools/docs/docs.go

```diff
@@ -76,25 +95,40 @@ func main() {
 	}
 }
 
-func findJsonSections(fileLines []string) [][]string {
+func findJsonSections(fileLines []string) ([][]string, error) {
 	var jsonSections [][]string
 	var currentSection []string
-	inASection := false
+
+	var state sectionState = notInSection
+
 	for _, line := range fileLines {
-		if line == "```" {
-			inASection = false
-			if len(currentSection) > 0 && currentSection[0] != "..." {
-				// Ignore empty sections and sections that are not full configs
-				jsonSections = append(jsonSections, currentSection)
+		switch state {
+		case notInSection:
+			if line == sectionMarker {
+				state = expectingSection
+			}
+
+		case expectingSection:
+			if line == "```json" {
+				state = inSection
+			} else {
+				return jsonSections, fmt.Errorf("expecting '```json', found: %s", line)
+			}
+
+		case inSection:
+			if line == "```" {
+				if len(currentSection) == 0 || currentSection[0] == "..." {
+					// Ignore empty sections and sections that are not full configs
+					fmt.Printf("Ignoring empty or partial config section")
```

Comment:
Do we need to report this?

Reply:
There currently aren't any, so I though it would be good to make sure we do not add some by mistake.

Reply:
Partial config sections are reasonable, which is why we have this special case.

Reply:
OK. I've moved this output to a file level summary of ignored/partial section instead.